jdaviz/core/template_mixin.py

```diff
@@ -230,32 +302,50 @@ def _update_has_subregions(self):
     def labels(self):
         return [s['label'] for s in self.items if 'label' in s.keys()]
 
+    @cached_property
```

Comment:
Do we need a way to invalidate cache if we start to allow people to rename stuff?